autogpt_platform/frontend/src/components/integrations/credentials-provider.tsx

```diff
@@ -173,6 +183,22 @@ export default function CredentialsProvider({
     [api, addCredentials],
   );
 
+  /** Wraps `BackendAPI.createHostScopedCredentials`, and adds the result to the internal credentials store. */
+  const createHostScopedCredentials = useCallback(
+    async (
+      provider: CredentialsProviderName,
+      credentials: HostScopedCredentialsCreatable,
+    ): Promise<CredentialsMetaResponse> => {
+      const credsMeta = await api.createHostScopedCredentials({
+        provider,
+        ...credentials,
+      });
+      addCredentials(provider, credsMeta);
+      return credsMeta;
+    },
+    [api, addCredentials],
+  );
```

Comment:
What happens if this API call fails? Are we handling the error on the `BackendAPI` client? We should at least show a toast 💭

Reply:
I have added the toast, I see all the api call here in this file don't call toast on failure, so I added each of them here.
Again, the PR gets bigger though, hehe